### PR TITLE
Adding  back the dataplane services

### PR DESCRIPTION
**Pull request by fao89**
Depends-On: https://review.rdoproject.org/r/c/rdo-jobs/+/50327

### CODE REVIEW

**Comment by fao89** [comment]
```
TASK [osp.edpm.edpm_iscsid : Download needed container] ************************
Thursday 28 September 2023  01:16:46 +0000 (0:00:00.194)       0:00:43.069 **** 
[1;30mFAILED - RETRYING: [standalone]: Download needed container (5 retries left).[0m
[1;30mFAILED - RETRYING: [standalone]: Download needed container (4 retries left).[0m
[1;30mFAILED - RETRYING: [standalone]: Download needed container (3 retries left).[0m
[1;30mFAILED - RETRYING: [standalone]: Download needed container (2 retries left).[0m
[1;30mFAILED - RETRYING: [standalone]: Download needed container (1 retries left).[0m
[0;31mfatal: [standalone]: FAILED! => {"attempts": 5, "changed": false, "msg": "Failed to pull image quay.io/podified-antelope-centos9/openstack-iscsid:current-podified"}[0m
```
https://logserver.rdoproject.org/71/171/1dc75617ef47d1f77c16a719b51120d928601f27/github-check/data-plane-adoption-github-rdo-centos-9-crc-single-node/0e8168e/controller/pod/dataplane-deployment-download-cache-openstack-m458k-logs.txt

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Review by jistr — APPROVED** [review]

**Comment by fao89** [comment]
@stuggi could you please review if I got the routes right?

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
it is getting stuck at:
```
TASK [osp.edpm.edpm_container_manage : Exclude Ceph containers from podman container list] ***
Saturday 30 September 2023  01:27:41 +0000 (0:00:02.531)       0:02:47.775 **** 
```
https://logserver.rdoproject.org/71/171/c064b38f5bf8eed510c691cf80d8b30fb53d5345/github-check/data-plane-adoption-github-rdo-centos-9-crc-single-node/2a32835/controller/pod/dataplane-deployment-run-os-openstack-p5lt4-logs.txt

@fultonj could you please take a look?

**Comment by fultonj** [comment]
> it is getting stuck at:
> 
> ```
> TASK [osp.edpm.edpm_container_manage : Exclude Ceph containers from podman container list] ***
> Saturday 30 September 2023  01:27:41 +0000 (0:00:02.531)       0:02:47.775 **** 
> ```
> 
> https://logserver.rdoproject.org/71/171/c064b38f5bf8eed510c691cf80d8b30fb53d5345/github-check/data-plane-adoption-github-rdo-centos-9-crc-single-node/2a32835/controller/pod/dataplane-deployment-run-os-openstack-p5lt4-logs.txt
> 
> @fultonj could you please take a look?

Please try again with this:

https://github.com/openstack-k8s-operators/edpm-ansible/pull/385

If it doesn't help, then let me know.

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fultonj** [comment]
> I tested locally, [Exclude Ceph containers from podman container](https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_container_manage/tasks/delete_orphan.yml#L23) is taking a long time, which is timing out the CI
> 
> ```
> osp.edpm.edpm_container_manage : Exclude Ceph containers from podman container list - 695.17s
> osp.edpm.edpm_container_manage : Exclude Ceph containers from podman container list - 414.45s
> osp.edpm.edpm_container_manage : Exclude Ceph containers from podman container list - 144.40s
> ```
> 
> full logs: https://paste.centos.org/view/e45553fe

This should unblock you:

https://github.com/openstack-k8s-operators/edpm-ansible/pull/387

It breaks HCI though so I'll need to reintroduce excluding the ceph containers but more efficiently.

**Comment by fao89** [comment]
I've started this: https://github.com/openstack-k8s-operators/edpm-ansible/pull/386

**Comment by fultonj** [comment]
> I've started this: [openstack-k8s-operators/edpm-ansible#386](https://github.com/openstack-k8s-operators/edpm-ansible/pull/386)

I've approved the above since it passed my tests in the HCI environment.

1. reproduce bug by doing revert
2. test your patch and see it not hit the same bug
3. log the `filtered_containers` generated by your patch and read that they have openstack containers but not ceph containers (as they should).

So hopefully you resolved your issue.

**Comment by fao89** [comment]
recheck

**Comment by cescgina** [comment]
recheck looks like a crc hiccup while creating the openstack namespace https://logserver.rdoproject.org/71/171/1f33c7ac59c8e996f8cf8b1f9ed6ae22531f586d/github-check/data-plane-adoption-github-rdo-centos-9-crc-single-node/6caedf3/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_minimal_out_2023-10-03T20:34:58EDT.log :
```
TASK [prelude_local : set up and use openstack namespace] **********************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -euxo pipefail\n\n\ncd /home/zuul/src/github.com/openstack-k8s-operators/install_yamls/\nmake namespace\n", "delta": "0:00:07.218662", "end": "2023-10-03 20:35:11.270343", "msg": "non-zero return code", "rc": 2, "start": "2023-10-03 20:35:04.051681", "stderr": "+ cd /home/zuul/src/github.com/openstack-k8s-operators/install_yamls/\n+ make namespace\n+ '[' -z /home/zuul/src/github.com/openstack-k8s-operators/install_yamls/out ']'\n+ '[' -z openstack ']'\n+ OUT_DIR=/home/zuul/src/github.com/openstack-k8s-operators/install_yamls/out/openstack\n+ '[' '!' -d /home/zuul/src/github.com/openstack-k8s-operators/install_yamls/out/openstack ']'\n+ mkdir -p /home/zuul/src/github.com/openstack-k8s-operators/install_yamls/out/openstack\n+ cat\nerror: unable to default to a user name: the server is currently unable to handle the request (get users.user.openshift.io ~)
```

**Comment by fao89** [comment]
> Build failed (check pipeline). Post `recheck` (without leading slash) to rerun all jobs. Make sure the failure cause has been resolved before you rerun jobs.
> 
> https://review.rdoproject.org/zuul/buildset/83df0d3837a848ada90689b64a494161
> 
> ❌ [data-plane-adoption-github-rdo-centos-9-crc-single-node ](https://review.rdoproject.org/zuul/build/0ebfa6b0735c470db61607e6b6377efd) **FAILURE** in 2h 13m 38s

ansible-runner image didn't get updated yet 😕

**Comment by cescgina** [comment]
> @cescgina does CI cache quay.io/openstack-k8s-operators/openstack-ansibleee-runner ? my PR was merged yesterday, but the image does seem to be updated

I don't think so, maybe the operator image needs to be updated to pull this new changes? At the moment in CI we install the operators by doing `make openstack` from install_yamls and as far as I can see there is nothing that would prevent it from pulling the latest image

**Comment by fao89** [comment]
recheck

**Comment by jistr** [comment]
🟢🟢🟢
🚀

File: tests/roles/dataplane_adoption/tasks/main.yaml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+- name: force latest ansible-runner image
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    {{ oc_login_command }}
+    oc get csv openstack-ansibleee-operator.v0.0.1 -o yaml -n openstack-operators | sed "s/RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT/AEE_IMAGE/" | oc apply -f -
```
Comment (fao89) [inline tests/roles/dataplane_adoption/tasks/main.yaml:7]: hack for usingi the latest ansible-runner image

**Pull request merged**